alt5 method-based syntax updates to all chapters

joelostblom opened the pull request:
closes #167

joelostblom committed:
Replace ggplot terminology

joelostblom committed:
Replace ggplot terminology

joelostblom committed:
Correct explanation of maxbins

joelostblom committed:
Correct explanation of maxbins

joelostblom committed:
Clarify language and use altair terminology

joelostblom committed:
Make use of parenthesis consistent within the chapter

joelostblom committed:
Explain alt.X and alt.Y and be consistent in using it only when there are no options passed

joelostblom committed:
Explain alt.X and alt.Y and be consistent in using it only when there are no options passed

joelostblom committed:
Correct "argument" with "method"

joelostblom committed:
Change intro and viz chapter to method based syntax

joelostblom committed:
Update classification chapters to method based syntax

joelostblom committed:
Remove git conflict indicators

joelostblom committed:
Update regression chapters to method based syntax

joelostblom committed:
Re-standardize syntax

Only needed because I didnnot pull new changes../..

joelostblom committed:
Add back scale zero=false

joelostblom committed:
Revert a couple of accidental changes

joelostblom committed:
Minor fixes and clarifications

joelostblom committed:
Updates to method syntax in classification 2

joelostblom committed:
Updates to method syntax in inference and regression

joelostblom committed:
Render histograms with appropriate bin widths

joelostblom commented on source/inference.md:
```diff
@@ -645,15 +645,23 @@ glue(
+                bin=alt.Bin(extent=[0, 660], maxbins=40),
                 title="Price per night (Canadian dollars)",
-                scale=alt.Scale(domainMax=700)
+                #scale=alt.Scale(domainMax=700)
             )
         ).properties(
             title='Population', height=150
         ),
-        sample_distribution.properties(title="Sample (n = 40)").properties(height=150),
-        sampling_distribution.properties(
+        sample_distribution.encode(
+            x=alt.X("price")
+                .bin(extent=[0, 660], maxbins=40)
+                .title("Price per night (Canadian dollars)")
+        ).properties(title="Sample (n = 40)").properties(height=150),
+        sampling_distribution.encode(
+            x=alt.X("sample_mean")
+                .bin(extent=[0, 660], maxbins=40)
+                .title("Price per night (Canadian dollars)")
+        ).properties(
```
This fix changes the appearance from uneven bins:

![image](https://github.com/UBC-DSCI/introduction-to-datascience-python/assets/4560057/0fff000a-a8b8-4e6d-b87d-d3bd2f11d264)

to even bins:

![image](https://github.com/UBC-DSCI/introduction-to-datascience-python/assets/4560057/ab6e5514-d161-4e7e-9214-fd369b383da2)

which is easier to compare (and more similar to the R version):

![image](https://github.com/UBC-DSCI/introduction-to-datascience-python/assets/4560057/fe18d86b-f5c4-4667-a02c-3c0e90bba4f6)

trevorcampbell replied:
nice catch!

joelostblom committed:
Add artefacts

trevorcampbell committed:
Merge pull request #183 from UBC-DSCI/main

Merge new dockerfile into alt5-chaining

trevorcampbell reviewed (changes requested):
Lots of great changes here! Some comments/fixes needed:

- Multiple figures in classification 1 aren't rendering any more, and broken figure references
- some bars in inference are too thick now (just one example here but there are many)

e.g.

![image](https://github.com/UBC-DSCI/introduction-to-datascience-python/assets/5981307/9ebcd603-f1aa-446c-90c7-ceaaee9d7d6d)

vs

![image](https://github.com/UBC-DSCI/introduction-to-datascience-python/assets/5981307/2be3fb1c-590f-4fb3-a020-59ed2bf19088)

- Some figure aspect ratios have changed (for the worse)
  - but some figures also seem to fit my browser window better now. not sure if this was an intentional change or if it's just a new default in altair / jupyter-book or something
  
- syntax highlighting seems to have changed, FYI, compared to the current public website. But I'm fine with it, it looks nice to me -- probably not related to anything you did, just the new docker image

![image](https://github.com/UBC-DSCI/introduction-to-datascience-python/assets/5981307/bee2d1c0-beab-4c12-9eff-adcc9ce570ca)

vs

![image](https://github.com/UBC-DSCI/introduction-to-datascience-python/assets/5981307/3bdb2071-0fea-43d3-bae4-dcbe40c3ae53)

joelostblom commented:
Thanks @trevorcampbell !

> Multiple figures in classification 1 aren't rendering any more, and broken figure references

Good catch, fixed I think. I only see warnings about missing citation keys when I build the HTML now, but I believe those were there previously:

```
/home/jovyan/source/viz.md:2021: WARNING: duplicate citation for key "mckinney2012python"
/home/jovyan/source/viz.md:2021: WARNING: duplicate citation for key "timbers2020canlang"
/home/jovyan/source/wrangling.md:1822: WARNING: duplicate citation for key "mckinney2012python"
/home/jovyan/source/classification2.md:1995: WARNING: duplicate citation for key "james2013introduction"
/home/jovyan/source/clustering.md:794: WARNING: duplicate citation for key "james2013introduction"
/home/jovyan/source/inference.md:1347: WARNING: duplicate citation for key "moderndive"
/home/jovyan/source/classification1.md:1889: WARNING: duplicate citation for key "knncover"
/home/jovyan/source/classification1.md:1889: WARNING: duplicate citation for key "knnfix"
/home/jovyan/source/classification1.md:1889: WARNING: duplicate citation for key "streetbreastcancer"
```


> some bars in inference are too thick now (just one example here but there are many)

Fixed by adding back `maxbins=30` for the charts where it was removed

> Some figure aspect ratios have changed (for the worse) but some figures also seem to fit my browser window better now. not sure if this was an intentional change or if it's just a new default in altair / jupyter-book or something.

In Altair 5 we changed the default from a 4:3 ratio to a square 3:3 ratio to avoid distortion in scatter plots and align with how things are done in the underlying Vega-Lite library. If there are specific figures where you think this is unhelpful we could set the width manually.

> syntax highlighting seems to have changed, FYI, compared to the current public website. But I'm fine with it, it looks nice to me -- probably not related to anything you did, just the new docker image

I noticed this as well and it looks OK to me too.

joelostblom committed:
Make sure all figures display and references resolve

trevorcampbell reviewed (approved):
lgtm
